portfolio/src/main/java/com/google/sps/servlets/ListDataServlet.java

```diff
@@ -37,11 +38,33 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment").addSort("message", SortDirection.DESCENDING);
+    int numCommentsToDisplay = getNumberOfComments(request);
+
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);
+    List<Entity> results = datastore.prepare(query).asList(FetchOptions.Builder.withLimit(numCommentsToDisplay));
+
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results) {
+        long id = entity.getKey().getId(); 
+        String name = (String) entity.getProperty("name");
+        String email = (String) entity.getProperty("email");
+        String subject = (String) entity.getProperty("subject");
+        String message = (String) entity.getProperty("message");
+        long timestamp = (long) entity.getProperty("timestamp");
+
+        Comment comment = new Comment(id, name, email, subject, message, timestamp);
+        comments.add(comment);
+    }
```

Comment:
Have you been introduced to the concept of Streams? We like to use them on the team since they're more declarative. It's not necessary, but If you feel like learning something new, try rewriting this with a stream.

https://docs.oracle.com/javase/8/docs/api/?java/util/stream/Stream.html